src/test/util.py

```diff
@@ -308,57 +287,65 @@ class wally_psbt(Structure):
     ('wally_pbkdf2_hmac_sha256', c_int, [c_void_p, c_ulong, c_void_p, c_ulong, c_uint, c_uint, c_void_p, c_ulong]),
     ('wally_pbkdf2_hmac_sha512', c_int, [c_void_p, c_ulong, c_void_p, c_ulong, c_uint, c_uint, c_void_p, c_ulong]),
+    ('wally_psbt_add_input_at', c_int, [POINTER(wally_psbt), c_uint, c_uint, POINTER(wally_tx_input)]),
+    ('wally_psbt_add_output_at', c_int, [POINTER(wally_psbt), c_uint, c_uint, POINTER(wally_tx_output)]),
     ('wally_psbt_combine', c_int, [POINTER(wally_psbt), POINTER(wally_psbt)]),
+    ('wally_psbt_clone_alloc', c_int, [POINTER(wally_psbt), c_uint, POINTER(POINTER(wally_psbt))]),
     ('wally_psbt_elements_init_alloc', c_int, [c_uint, c_ulong, c_ulong, c_ulong, POINTER(POINTER(wally_psbt))]),
+    ('wally_psbt_extract', c_int, [POINTER(wally_psbt), POINTER(POINTER(wally_tx))]),
+    ('wally_psbt_finalize', c_int, [POINTER(wally_psbt)]),
+    ('wally_psbt_free', c_int, [POINTER(wally_psbt)]),
+    ('wally_psbt_from_base64', c_int, [c_char_p, POINTER(POINTER(wally_psbt))]),
+    ('wally_psbt_from_bytes', c_int, [c_void_p, c_ulong, POINTER(POINTER(wally_psbt))]),
+    ('wally_psbt_get_length', c_int, [POINTER(wally_psbt), c_uint, c_ulong_p]),
+    ('wally_psbt_init_alloc', c_int, [c_uint, c_ulong, c_ulong, c_ulong, POINTER(POINTER(wally_psbt))]),
+    ('wally_psbt_input_add_keypath_item', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong, c_void_p, c_ulong, c_uint_p, c_ulong]),
+    ('wally_psbt_input_add_signature', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong, c_void_p, c_ulong]),
     ('wally_psbt_input_clear_value', c_int, [POINTER(wally_psbt_input)]),
+    ('wally_psbt_input_find_keypath', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong, c_ulong_p]),
+    ('wally_psbt_input_find_signature', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong, c_ulong_p]),
+    ('wally_psbt_input_find_unknown', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong, c_ulong_p]),
+    ('wally_psbt_input_is_finalized', c_int, [POINTER(wally_psbt_input), c_ulong_p]),
     ('wally_psbt_input_set_abf', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong]),
     ('wally_psbt_input_set_asset', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong]),
     ('wally_psbt_input_set_claim_script', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong]),
+    ('wally_psbt_input_set_final_scriptsig', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong]),
+    ('wally_psbt_input_set_final_witness', c_int, [POINTER(wally_psbt_input), POINTER(wally_tx_witness_stack)]),
     ('wally_psbt_input_set_genesis_blockhash', c_int, [POINTER(wally_psbt_input), c_void_p, c_ulong]),
-    ('wally_psbt_input_set_peg_in_tx', c_int, [POINTER(wally_psbt_input), POINTER(wally_tx)]),
+    ('wally_psbt_input_set_keypaths', c_int, [POINTER(wally_psbt_input), POINTER(wally_map)]),
+    ('wally_psbt_input_set_utxo', c_int, [POINTER(wally_psbt_input), POINTER(wally_tx)]),
```

Comment:
Lines generated with `./tools/build_python_cffi.sh` are in a different order w.r.t the ones in `src/test/util.py`. We may want to have them consistent.

Reply:
Yes, the ones in the file are sorted by vim :%sort which differs from /usr/bin/sort.

The current sed/perl generation approach is not sufficiently general to be used for generating wally.hpp, or the swig python/java required typemaps, all of which are constantly out of date as was the cffi list here. So my plan is to write a python parser which can generate all of these and have it write the new values in place. That will resolve the issue of sorting since there will be no manual changes to this list following that. Until then I'm OK with updating this manually and living with the sort difference.

Reply:
I've raised https://github.com/ElementsProject/libwally-core/issues/212 for this.